app/controllers/concerns/idv_session.rb

```diff
@@ -12,12 +12,52 @@ def confirm_idv_session_started
     redirect_to idv_session_url unless idv_session[:params].present?
   end
 
+  def confirm_idv_attempts_allowed
+    if idv_attempts_exceeded?
+      flash[:error] = t('idv.errors.hardfail')
+      redirect_to idv_fail_url
+    end
+  end
+
   def idv_session
     user_session[:idv] ||= {}
   end
 
   protected
 
+  def idv_max_attempts
+    (Figaro.env.idv_max_attempts || 3).to_i
+  end
+
+  def idv_attempt_window
+    (Figaro.env.idv_attempt_window || 24).to_i.hours
+  end
+
+  def idv_attempts=(num)
+    user_session[:idv_attempts] = num
+  end
+
+  def idv_attempts
+    user_session[:idv_attempts] ||= 0
+  end
+
+  def idv_attempts_exceeded?
+    # too many attempts in this session
+    return true if idv_attempts >= idv_max_attempts
+
+    # never attempted
+    attempted_at = current_user.idv_attempted_at
+    return false unless attempted_at.present?
+
+    # recent attempts in any session
+    attempted_at + idv_attempt_window > Time.zone.now
+  end
```

Comment:
thanks @amoose -- I now have the browser extension as well. I added a test and it looks 💚  to me -- can you confirm?